packages/strapi-admin/admin/src/containers/Webhooks/EditView/tests/reducer.test.js

```diff
@@ -0,0 +1,255 @@
+      .setIn(['initialData', 'headers'], fromJS(initialHeaders))
+      .setIn(['modifiedData', 'headers'], fromJS(initialHeaders));
+
+    const action = {
+      type: 'ON_HEADER_REMOVE',
+      index: 1,
+    };
+
+    const updatedHeaders = [{ key: 'accept', value: 'text/html' }];
+
+    const expectedState = state.setIn(
+      ['modifiedData', 'headers'],
+      fromJS(updatedHeaders)
+    );
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should clear a header to modifiedData on ON_HEADER_REMOVE if header is the last', () => {
+    const initialHeaders = [{ key: 'accept', value: 'text/html' }];
+    const state = initialState
+      .setIn(['initialData', 'headers'], fromJS(initialHeaders))
+      .setIn(['modifiedData', 'headers'], fromJS(initialHeaders));
+
+    const action = {
+      type: 'ON_HEADER_REMOVE',
+      index: 0,
+    };
+
+    const updatedHeaders = [header];
+
+    const expectedState = state.setIn(
+      ['modifiedData', 'headers'],
+      fromJS(updatedHeaders)
+    );
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should update isTriggering on ON_TRIGGER_CANCELED', () => {
+    const state = initialState.set('isTriggering', true);
+
+    const action = {
+      type: 'ON_TRIGGER_CANCELED',
+    };
+
+    const expectedState = state.update('isTriggering', () => false);
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should clear triggerResponse on ON_TRIGGER_CANCELED', () => {
+    const state = initialState.set('triggerResponse', {
+      statusCode: 200,
+      message: 'succeed',
+    });
+
+    const action = {
+      type: 'ON_TRIGGER_CANCELED',
+    };
+
+    const expectedState = state.update('triggerResponse', () => fromJS({}));
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should reset modifiedData with initialData values on RESET_FORM', () => {
+    const state = initialState
+      .setIn(['modifiedData', 'name'], 'updated name')
+      .setIn(['modifiedData', 'url'], 'updated url');
+
+    const action = {
+      type: 'RESET_FORM',
+    };
+
+    const expectedState = state.update('modifiedData', () =>
+      state.get('initialData')
+    );
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should update formErrors with the right params on SET_ERRORS', () => {
+    const state = initialState;
+
+    const action = {
+      type: 'SET_ERRORS',
+      errors: {
+        name: 'The value is required',
+      },
+    };
+
+    const expectedState = state.update('formErrors', () =>
+      fromJS(action.errors)
+    );
+
+    expect(reducer(state, action)).toEqual(expectedState);
+  });
+
+  it('should toggle isTriggering on SET_IS_TRIGGERING', () => {
+    const state = initialState;
+
+    const action = {
+      type: 'SET_IS_TRIGGERING',
+    };
+
+    const expectedState = state.update(
+      'isTriggering',
+      isTriggering => !isTriggering
```

Comment:
Here your test is the exact same code than the code you are trying to test. You should use `setIn` instead of `updateIn` so you are sure your logic is correct.